src/promptflow/promptflow/_sdk/entities/_connection.py

```diff
@@ -346,10 +347,12 @@ class AzureOpenAIConnection(_StrongTypeConnection):
     TYPE = ConnectionType.AZURE_OPEN_AI
 
     def __init__(
-        self, api_key: str, api_base: str, api_type: str = "azure", api_version: str = "2023-07-01-preview", **kwargs
+        self, api_key: str, api_base: str, api_type: str = "azure", api_version: str = "2023-07-01-preview",
+        token_provider: Union[str, TokenProviderABC] = AzureTokenProvider(), **kwargs
```

Comment:
Usually it would be better to have default as None and create the instance in the init function.
Otherwise a AzureTokenProvider instance would be used in different connection instances.